Runtime/Scripts/Export/ImageExport.cs

```diff
@@ -64,9 +64,33 @@ public ImageExport(Texture2D texture, ImageFormat imageFormat = ImageFormat.Unkn
             m_ImageFormat = imageFormat;
 #if UNITY_EDITOR
             m_AssetPath = AssetDatabase.GetAssetPath(texture);
+            var importer = (TextureImporter)TextureImporter.GetAtPath(m_AssetPath);
+            if (importer != null && importer.alphaSource == TextureImporterAlphaSource.FromGrayScale && importer.alphaIsTransparency)
+            {
+                CreateTransparentTexture(texture);
+            }
```

Comment:
This part is indeed kind of hacky. Not sure if the transparent texture handling can be moved to `GenerateTexture()` and `Write()` instead.

Currently I think it's still fine to have special handling for one particular type of texture, but if we have more special cases, it could get unwieldy if we handle each one separately.

Reply:
Yeah it could be moved to the other methods, not sure which way is better 🤔 

> Currently I think it's still fine to have special handling for one particular type of texture, but if we have more special cases, it could get unwieldy if we handle each one separately.

I feel like there wouldn't be too many special cases for textures, if I'm not mistaken, I think handling more special cases of shaders could become problematic down the road if anything.

Reply:
We can start refactoring if we have more special cases.

Yes, actually a lot of these are work-arounds, because the materials are not standardised. Unity alone has 3 different render pipelines, and a lot of materials don't work across the different pipelines. If we want seamless support of Unity materials, it probably warrants a long discussion. Serialising everything doesn't seem to be a good solution :sweat_smile:

Currently I think we handle the base case (standard shaders) and some common special cases, so it shouldn't get out of hand yet.